qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -0,0 +1,662 @@
+            value is calculated assuming a single range change will occur 
+            for every measurement (not multiple ranges, just one range up or 
+            down per measurement)."""))
+
+
+class Display(InstrumentChannel):
+    """Implements interaction with the display of Keysight 344xxA."""
+
+    def __init__(self, parent: '_Keysight_344xxA', name: str, **kwargs):
+        super(Display, self).__init__(parent, name, **kwargs)
+
+        self.add_parameter('enabled',
+                           label='Display enabled',
+                           set_cmd='DISPlay:STATe {}',
+                           get_cmd='DISPlay:STATe?',
+                           val_mapping={True: 1, False: 0},
+                           docstring=textwrap.dedent("""\
+            Disables or enables the front panel display. When disabled, 
+            the display dims, and all annunciators are disabled. However, 
+            the screen remains on.
+
+            Disabling the display improves command execution speed from the 
+            remote interface and provides basic security.
+
+            Displaying text with `display.text` parameter will work even 
+            when the display is disabled."""))
+
+        self.add_parameter('text',
+                           label='Display text',
+                           set_cmd='DISPLAY:TEXT "{}"',
+                           get_cmd='DISPLAY:TEXT?',
+                           get_parser=lambda s: s.strip('"'),
+                           vals=vals.Strings(),
+                           docstring=textwrap.dedent("""\
+            Displays the given text on the screen. Specifying empty string 
+            moves the display back to its normal state. The same can be 
+            achieved by calling `display.clear`."""))
+
+    def clear(self) -> None:
+        """
+        Clear text from display. Depending on the display being
+        enabled/disabled, this either returns to display's normal state or
+        leaves it black, respectively.
+        """
+        self.write('DISPLay:TEXT:CLEar')
+        self.text.get()  # also update the parameter value
+
+
+class _Keysight_344xxA(KeysightErrorQueueMixin, VisaInstrument):
+    """
+    Instrument class for Keysight 34460A, 34461A, 34465A and 34470A
+    multimeters.
+
+    The driver currently only supports using the instrument as a voltmeter
+    for DC measurements.
+
+    This driver makes use of submodules for implementing different
+    subsystems of the instrument.
+
+    Attributes:
+        model (str): The model number of the instrument
+        NPLC_list (list): A list of the available Power Line Cycle settings
+        ranges (list): A list of the available voltage ranges
+    """
+
+    def __init__(self, name, address, silent=False,
+                 **kwargs):
+        """
+        Create an instance of the instrument.
+
+        Args:
+            name (str): Name used by QCoDeS. Appears in the DataSet
```

Comment:
is it on purpose that the types are still here and not as type hints in the signature?

Reply:
definitely not. i'll clean this up.